nntrainer/models/neuralnet.cpp

```diff
@@ -342,7 +341,7 @@ void NeuralNetwork::backwarding(int iteration,
     model_graph.flushCacheExcept(std::get<2>(node->getExecutionOrder()));
     PROFILE_MEM_ANNOTATE("CalcDerivative: " + node->getName());
 
-    if (stop_cb(nullptr)) {
+    if (stop_cb(nullptr, nullptr)) {
```

Comment:
I don't think we set the user data as nullptr by ourselves. How about giving more flexibility to users by providing enum where the stop call back is called?
``` c++
enum class RunStopCallback{
   AFTER_ITERARTION,
   AFTER_EPOCH,
   AFTER_EPOCH_EVAL,
   FORWARDING_EVERY_LAYER,
   BACKWARDING_EVERY_LAYER,
}
```

Reply:
Added StopCallbackCheckpointType enum class